From e131d64f6ebaf299267865f853c68ccc943a0ad1 Mon Sep 17 00:00:00 2001
From: Vasanthakumar K <vasanthakumark@ami.com>
Date: Wed, 18 Jun 2025 16:29:37 +0530
Subject: [PATCH] security-manager Removed Werror to not treat warnings as
 error

Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1c50703..3623780 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,6 @@ cmake_minimum_required (VERSION 3.6 FATAL_ERROR)
 project (security-manager CXX)
 
 set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -lstdc++fs \
-    -Werror \
     -Wall \
     -Wextra \
     -Wshadow \
-- 
2.34.1

