From 998cb60c59c32551b04741c4121e0a3a81e6a9b1 Mon Sep 17 00:00:00 2001
From: arjuncr <arjuncr@ami.com>
Date: Sun, 22 Feb 2026 08:08:21 -0500
Subject: [PATCH] Fix for 2700dcscm build error

---
 src/security-manager.cpp | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/security-manager.cpp b/src/security-manager.cpp
index ef0fa81..98476c5 100644
--- a/src/security-manager.cpp
+++ b/src/security-manager.cpp
@@ -443,10 +443,10 @@ static void fileShadowChangeHandler()
 
     std::array<char, 4096> buffer{};
     struct inotify_event* event = NULL;
-    int index = 0;
+    ssize_t index = 0;
     bool exec = 0;
 
-    int len = read(inotifyFd, buffer.data(), sizeof(buffer));
+    ssize_t len = read(inotifyFd, buffer.data(), sizeof(buffer));
     if (len < 0)
     {
         std::cerr << "fileChangeHandler length error: " << len;
@@ -469,9 +469,9 @@ static void fileShadowChangeHandler()
             addInotifyWatch(inotifyFd, "/etc/shadow");
         }
         /* Move to next struct */
-        len -= sizeof(inotify_event) + event->len;
+        len -= static_cast<ssize_t>(sizeof(inotify_event) + event->len);
         if (len > 0)
-            index += (sizeof(inotify_event) + event->len);
+            index += static_cast<ssize_t>(sizeof(inotify_event) + event->len);
         else
             event = NULL;
     }
@@ -507,10 +507,10 @@ static void filePwdChangeHandler()
 
     std::array<char, 4096> buffer{};
     struct inotify_event* event = NULL;
-    int index = 0;
+    ssize_t index = 0;
     bool exec = 0;
 
-    int len = read(inotifyPwdFd, buffer.data(), sizeof(buffer));
+    ssize_t len = read(inotifyPwdFd, buffer.data(), sizeof(buffer));
     if (len < 0)
     {
         std::cerr << "filePwdChangeHandler length error: " << len;
@@ -533,9 +533,9 @@ static void filePwdChangeHandler()
             addInotifyWatch(inotifyPwdFd, "/etc/passwd");
         }
         /* Move to next struct */
-        len -= sizeof(inotify_event) + event->len;
+        len -= static_cast<ssize_t>(sizeof(inotify_event) + event->len);
         if (len > 0)
-            index += (sizeof(inotify_event) + event->len);
+            index += static_cast<ssize_t>(sizeof(inotify_event) + event->len);
         else
             event = NULL;
     }
-- 
2.34.1

