From d3313dd904faf3b497157878322d2ef905f71ad2 Mon Sep 17 00:00:00 2001
From: rajamaickems <rajamanickams@ami.com>
Date: Mon, 21 Jul 2025 21:30:10 +0530
Subject: [PATCH 290/290] Fix for 2700 DCSCM BHS build

---
 CMakeLists.txt | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c58166d..a17e8ab 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,13 +23,10 @@ set (
     -Wunused \
     -Woverloaded-virtual \
     -Wpedantic \
-    -Wconversion \
     -Wmisleading-indentation \
     -Wduplicated-cond \
     -Wduplicated-branches \
     -Wlogical-op \
-    -Wnull-dereference \
-    -Wuseless-cast \
     -Wdouble-promotion \
     -Wno-strict-aliasing \
     -Wno-cast-align \
-- 
2.25.1

