diff --git a/target/jtag_handler.c b/target/jtag_handler.c
index c7afbb2..3fc6fa5 100644
--- a/target/jtag_handler.c
+++ b/target/jtag_handler.c
@@ -128,7 +128,7 @@ STATUS JTAG_initialize(JTAG_Handler* state, bool sw_mode)
 #ifdef JTAG_LEGACY_DRIVER
     state->JTAG_driver_handle = open("/dev/jtag", O_RDWR);
 #else
-    state->JTAG_driver_handle = open("/dev/jtag0", O_RDWR);
+    state->JTAG_driver_handle = open("/dev/jtag1", O_RDWR);
 #endif
     if (state->JTAG_driver_handle == -1)
     {
