diff --git a/target/asd_msg.c b/target/asd_msg.c
index 112bc5f..d675f1f 100644
--- a/target/asd_msg.c
+++ b/target/asd_msg.c
@@ -1263,15 +1263,26 @@ STATUS process_jtag_message(struct asd_message* s_message)
                     for (int i = chain_bytes_length - 1; i >= 0; i--)
                     {
                         size_t remaining_size = sizeof(line) - pos;
-                        snprintf(&line[pos], remaining_size, "%02X",
-                                 chain_bytes[i]);
-                        pos = pos + 2;
-                        if (i % 2 == 0 && i != 0)
+
+                        if (remaining_size > 2)
+                        {
+                            snprintf(&line[pos], remaining_size, "%02X",
+                                     chain_bytes[i]);
+                            pos = pos + 2;
+                        }
+
+                        remaining_size = sizeof(line) - pos;
+
+                        if (i % 2 == 0 && i != 0 && remaining_size > 1)
                         {
-                            remaining_size = sizeof(line) - pos;
                             snprintf(&line[pos], remaining_size, " ");
                             pos++;
                         }
+
+                        if (pos < sizeof(line))
+                        {
+                            line[pos] = '\0';
+                        }
                     }
 #ifdef ENABLE_DEBUG_LOGGING
                     ASD_log(ASD_LogLevel_Debug, ASD_LogStream_SDK,
@@ -2632,7 +2643,17 @@ STATUS asd_msg_read(void)
             size_t num_to_read =
                 sizeof(msg->header) - msg_state.in_msg.read_index;
 
-            size_t size = asd_api_server_read(buffer, num_to_read, NULL);
+            size_t header_size = sizeof(msg->header);
+            size_t size = 0;
+
+            if (header_size > 0)
+            {
+                buffer = (void*)((char*)&(msg->header) +
+                                 msg_state.in_msg.read_index);
+                num_to_read = header_size - msg_state.in_msg.read_index;
+            }
+
+            size = asd_api_server_read(buffer, num_to_read, NULL);
 
             if (size > 0)
             {
