From e450eaf18641ec0827ff97c678fcd9758eb3cd75 Mon Sep 17 00:00:00 2001
From: arjuncr <arjuncr@ami.com>
Date: Wed, 25 Feb 2026 11:55:46 -0500
Subject: [PATCH] mctp-emulator fix for LF sync

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fc10ac5..bb9bfd4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,7 +25,7 @@ set (
     -Wduplicated-cond \
     -Wduplicated-branches \
     -Wlogical-op \
-    -Wnull-dereference \
+    -Wno-null-dereference \
     -Wuseless-cast \
     -Wdouble-promotion \
     -Wformat=2 \
-- 
2.34.1

