From 8a1b036fcab0d022ad56d843cb911d8448d147cc Mon Sep 17 00:00:00 2001
From: arjuncr <arjuncr@ami.com>
Date: Thu, 19 Feb 2026 11:16:35 -0500
Subject: [PATCH] Fix For LF Sync Build Failure

---
 CMakeLists.txt | 2 +-
 mctpw.cpp      | 5 +++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7dab2bf..49e9e02 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,7 +27,7 @@ set (
     -Wduplicated-cond \
     -Wduplicated-branches \
     -Wlogical-op \
-    -Wnull-dereference \
+    -Wno-null-dereference \
     -Wuseless-cast \
     -Wdouble-promotion \
     -Wno-strict-aliasing \
diff --git a/mctpw.cpp b/mctpw.cpp
index db65653..3c9ce0b 100755
--- a/mctpw.cpp
+++ b/mctpw.cpp
@@ -21,6 +21,7 @@
 #include <stdlib.h>
 
 #include <atomic>
+#include <tuple>
 #include <boost/algorithm/string/classification.hpp>
 #include <boost/algorithm/string/split.hpp>
 #include <boost/asio/spawn.hpp>
@@ -817,7 +818,7 @@ int mctpw_async_send_message(void* client_context, mctpw_eid_t dst_eid,
     {
         clientContext* ctx = static_cast<clientContext*>(client_context);
 
-        boost::asio::spawn(
+        std::ignore = boost::asio::spawn(
             *(ctx->io_context),
             [ctx, user_ctx, dst_eid, tag_owner, tag, payload, payload_length,
              cb](boost::asio::yield_context yield) {
@@ -943,7 +944,7 @@ int mctpw_async_send_receive_atomic_message(
     try
     {
         clientContext* ctx = static_cast<clientContext*>(client_context);
-        boost::asio::spawn(
+        std::ignore = boost::asio::spawn(
             *(ctx->io_context),
             [ctx, user_ctx, dst_eid, request_payload, request_payload_length,
              timeout, cb](boost::asio::yield_context yield) {
-- 
2.34.1

