From 3f5aded518c280fc2f27e33987f727199effec61 Mon Sep 17 00:00:00 2001
From: Przemyslaw Czarnowski <przemyslaw.hawrylewicz.czarnowski@intel.com>
Date: Fri, 22 Apr 2022 10:40:56 +0200
Subject: [PATCH] Revert "Disable nbd proxy from the build"

Signed-off-by: Przemyslaw Czarnowski <przemyslaw.hawrylewicz.czarnowski@intel.com>
Upstream-Status: Pending
---
 meson.build       |  2 +-
 meson_options.txt | 10 +++++-----
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/meson.build b/meson.build
index ddb8b2c5..eaaf6446 100644
--- a/meson.build
+++ b/meson.build
@@ -95,7 +95,7 @@ feature_map = {
   'experimental-redfish-multi-computer-system'  : '-DBMCWEB_ENABLE_MULTI_COMPUTERSYSTEM',
   'vm-websocket'                                : '-DBMCWEB_ENABLE_VM_WEBSOCKET',
   'xtoken-auth'                                 : '-DBMCWEB_ENABLE_XTOKEN_AUTHENTICATION',
-  #'vm-nbdproxy'                                : '-DBMCWEB_ENABLE_VM_NBDPROXY',
+  'vm-nbdproxy'                                 : '-DBMCWEB_ENABLE_VM_NBDPROXY',
 }
 
 # Get the options status and build a project summary to show which flags are
diff --git a/meson_options.txt b/meson_options.txt
index c2aa7d75..baeff269 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -35,11 +35,11 @@ option(
 # this interface, so for the moment this appears to be dead code;  In leiu of
 # removing it, it has been disabled to try to give those that use it the
 # opportunity to upstream their backend implementation
-#option(
-#    'vm-nbdproxy',
-#    type: 'feature', value: 'disabled',
-#    description: 'Enable the Virtual Media WebSocket.'
-#)
+option(
+    'vm-nbdproxy',
+    type: 'feature', value: 'disabled',
+    description: 'Enable the Virtual Media WebSocket.'
+)
 
 option(
     'rest',
-- 
2.17.1

