From f9f41ffa6ea39bfd42ac7b029de984cdc7baa290 Mon Sep 17 00:00:00 2001
From: Vasanthakumar K <vasanthakumark@ami.com>
Date: Fri, 1 Aug 2025 17:13:46 +0530
Subject: [PATCH] hsbp-manager-Removed-Werror-to-not-treat-warnings-as-error

Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c41085d..56a863f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,6 @@ project (hsbp-manager CXX)
 set (
     CMAKE_CXX_FLAGS
     "${CMAKE_CXX_FLAGS} -lstdc++fs \
-    -Werror \
     -Wall \
     -Wextra \
     -Wshadow \
--
2.34.1

