From 93dd13286900a4ecb8cddd40b27a948cd021b867 Mon Sep 17 00:00:00 2001
From: "Kondiboina, VengaiahX" <vengaiahx.kondiboina@intel.com>
Date: Mon, 7 Aug 2023 10:26:08 +0000
Subject: [PATCH] Adding patch for smbios-mdr for defaultTimeout

Signed-off-by: Kondiboina, VengaiahX <vengaiahx.kondiboina@intel.com>
---
 include/smbios_mdrv2.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/smbios_mdrv2.hpp b/include/smbios_mdrv2.hpp
index c38b5ad..54334f5 100644
--- a/include/smbios_mdrv2.hpp
+++ b/include/smbios_mdrv2.hpp
@@ -45,7 +45,7 @@ constexpr uint32_t smbiosTableTimestamp = 0x45464748;
 constexpr uint32_t smbiosSMMemoryOffset = 0;
 constexpr uint32_t smbiosSMMemorySize = 1024 * 1024;
 constexpr uint32_t smbiosTableStorageSize = 64 * 1024;
-constexpr uint32_t defaultTimeout = 20000;
+constexpr uint32_t defaultTimeout = 2000000;
 
 enum class MDR2SMBIOSStatusEnum
 {
-- 
2.17.1

