diff --git a/lib/pkgconfig/modern-cpp-kafka.pc b/lib/pkgconfig/modern-cpp-kafka.pc
new file mode 100644
index 0000000..389dba2
--- /dev/null
+++ b/lib/pkgconfig/modern-cpp-kafka.pc
@@ -0,0 +1,10 @@
+prefix=/usr
+exec_prefix=${prefix}
+includedir=${prefix}/include
+libdir=${prefix}/lib
+
+Name: modern-cpp-kafka
+Description: C++ wrapper on librdkafka
+Version: 2021.12.08
+Requires: rdkafka
+Cflags: -I${includedir}
