From e7df0e7c424c86d7caaa4c316d724761c0ef299a Mon Sep 17 00:00:00 2001
From: arjuncr <arjuncr@ami.com>
Date: Mon, 11 May 2026 05:14:51 -0400
Subject: [PATCH] Fix for Uninitialized Variable

---
 afmctp-stack/mctp_impl.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/afmctp-stack/mctp_impl.cpp b/afmctp-stack/mctp_impl.cpp
index 9af089d..8899299 100644
--- a/afmctp-stack/mctp_impl.cpp
+++ b/afmctp-stack/mctp_impl.cpp
@@ -602,8 +602,8 @@ std::pair<boost::system::error_code, ByteArray>
     }
 
     boost::system::error_code ec;
-    socklen_t addrlen;
     struct sockaddr_mctp addr{0};
+    socklen_t addrlen = sizeof(addr);
     addr.smctp_family = AF_MCTP;
     addr.smctp_network = devID.networkId();
     addr.smctp_addr.s_addr = devID.mctpEID();
-- 
2.34.1

