From 38879284c419ee68c3a155e813e018082d077fa9 Mon Sep 17 00:00:00 2001
From: leon_lin <leonlin@ami.com>
Date: Thu, 22 Jan 2026 11:57:52 +0800
Subject: [PATCH 1/2] Added OpenBMC GCC toolchain, headers, export library

---
 CMakeLists.txt | 47 +++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 47 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9909b25b..84c4a43c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -106,6 +106,8 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
         message("TOOLCHAIN = ARC_GCC")
     elseif(TOOLCHAIN STREQUAL "NIOS2_GCC")
         message("TOOLCHAIN = NIOS2_GCC")
+    elseif(TOOLCHAIN STREQUAL "OPENBMC_GCC")
+        message("TOOLCHAIN = OPENBMC_GCC")
     elseif(TOOLCHAIN STREQUAL "NONE")
         message("TOOLCHAIN = NONE")
     else()
@@ -634,6 +636,40 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
         set(CMAKE_EXE_LINKER_FLAGS "")
 
         set(CMAKE_C_LINK_EXECUTABLE "")
+    elseif(TOOLCHAIN STREQUAL "OPENBMC_GCC")
+        set(CMAKE_C_COMPILER arm-openbmc-linux-gnueabi-gcc)
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99 -g  -fno-strict-aliasing -Wall -Werror -Wno-array-bounds -ffunction-sections -fdata-sections -fno-common -Wno-address -fPIC -fno-asynchronous-unwind-tables -Wno-maybe-uninitialized -Wno-uninitialized  -Wno-builtin-declaration-mismatch -Wno-nonnull-compare")
+        if(NOT CMAKE_BUILD_TYPE STREQUAL "Release")
+            add_compile_options(-g)
+            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-error=cpp")
+        endif()
+        if(GCOV STREQUAL "ON")
+            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} --coverage -fprofile-arcs -ftest-coverage")
+        endif()
+        set(MBEDTLS_FLAGS "")
+        set(OPENSSL_FLAGS -include base.h -Wno-error=maybe-uninitialized -Wno-error=format -Wno-format -Wno-format-security -Wno-error=unused-but-set-variable -Wno-cast-qual)
+        set(CMOCKA_FLAGS -std=gnu99 -Wpedantic -Wall -Wshadow -Wmissing-prototypes -Wcast-align -Werror=address -Wstrict-prototypes -Werror=strict-prototypes -Wwrite-strings -Werror=write-strings -Werror-implicit-function-declaration -Wpointer-arith -Werror=pointer-arith -Wdeclaration-after-statement -Werror=declaration-after-statement -Wreturn-type -Werror=return-type -Wuninitialized -Werror=uninitialized -Werror=strict-overflow -Wstrict-overflow=2 -Wno-format-zero-length -Wmissing-field-initializers -Wformat-security -Werror=format-security -fno-common -Wformat -fno-common -fstack-protector-strong -Wno-cast-qual)
+        set(CMAKE_AR arm-openbmc-linux-gnueabi-gcc-ar)
+        if(NOT CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 9.3)
+            set(CMAKE_C_ARCHIVE_FINISH true)
+        endif()
+
+        set(CMAKE_LINKER arm-openbmc-linux-gnueabi-gcc)
+        set(CMAKE_EXE_LINKER_FLAGS "$ENV{LDFLAGS} ${OBMCGCCOPTION} -flto -Wno-error -no-pie" )
+        if(GCOV STREQUAL "ON")
+            set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS}   --coverage -lgcov -fprofile-arcs -ftest-coverage")
+        endif()
+        if(ENABLE_BINARY_BUILD STREQUAL "1")
+            set(CMAKE_C_LINK_EXECUTABLE "<CMAKE_LINKER> <LINK_FLAGS> <OBJECTS> -o <TARGET> -Wl,--start-group <LINK_LIBRARIES> -lpthread -ldl -Wl,--end-group")
+        else()
+            set(CMAKE_C_LINK_EXECUTABLE "<CMAKE_LINKER> <LINK_FLAGS> <OBJECTS> -o <TARGET> -Wl,--start-group <LINK_LIBRARIES> -Wl,--end-group")
+        endif()
+        if(FIPS_MODE STREQUAL "1")
+            message("FIPS Enabled!!.")
+            add_definitions(-DLIBSPDM_FIPS_MODE=1)
+        else()
+            message("FIPS Disabled.")
+        endif()
     endif()
 
     if(NOT TOOLCHAIN STREQUAL "NIOS2_GCC")
@@ -1130,3 +1166,14 @@ else()
         install(TARGETS ${LIB_NAME}_crypto LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
     endif()
 endif()
+
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/lib/libspdm.a DESTINATION /usr/lib)
+    install(DIRECTORY include DESTINATION /usr/include/libspdm)
+    install(DIRECTORY os_stub/include DESTINATION /usr/include/libspdm/os_stub)
+    install(FILES os_stub/spdm_crypt_ext_lib/spdm_crypt_ext_lib.h DESTINATION /usr/include/libspdm/os_stub/spdm_crypt_ext_lib)
+    install(FILES os_stub/spdm_crypt_ext_lib/cryptlib_ext.h DESTINATION /usr/include/libspdm/os_stub/spdm_crypt_ext_lib)
+    install(FILES os_stub/spdm_device_secret_lib_sample/raw_data_key.h DESTINATION /usr/include/libspdm/os_stub)
+    install(FILES os_stub/spdm_device_secret_lib_sample/spdm_device_secret_lib_internal.h DESTINATION /usr/include/libspdm/os_stub)
+    add_library(spdm STATIC ${SPDM_OBJECTS})
+    INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/lib/pkgconfig/libspdm.pc DESTINATION /usr/lib/pkgconfig)
+    CONFIGURE_FILE(libspdm.pc.in lib/pkgconfig/libspdm.pc @ONLY)
-- 
2.34.1

