From a04e130879ea9d6a5e1ecfdfbc3620cc62b073ca Mon Sep 17 00:00:00 2001
From: eddyhan <eddyhan@ami.com>
Date: Mon, 15 Jul 2024 20:33:53 +0800
Subject: [PATCH 2/2] create libspdm


diff --git a/library/spdm_common_lib/CMakeLists.txt b/library/spdm_common_lib/CMakeLists.txt
index a42fc703cd..1629db7528 100644
--- a/library/spdm_common_lib/CMakeLists.txt
+++ b/library/spdm_common_lib/CMakeLists.txt
@@ -13,3 +13,5 @@ SET(src_spdm_common_lib
 )
 
 ADD_LIBRARY(spdm_common_lib STATIC ${src_spdm_common_lib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:spdm_common_lib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/library/spdm_crypt_lib/CMakeLists.txt b/library/spdm_crypt_lib/CMakeLists.txt
index 6ef0511f47..69b442e0b3 100644
--- a/library/spdm_crypt_lib/CMakeLists.txt
+++ b/library/spdm_crypt_lib/CMakeLists.txt
@@ -27,3 +27,5 @@ SET(src_spdm_crypt_lib
 )
 
 ADD_LIBRARY(spdm_crypt_lib STATIC ${src_spdm_crypt_lib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:spdm_crypt_lib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/library/spdm_requester_lib/CMakeLists.txt b/library/spdm_requester_lib/CMakeLists.txt
index 31d4e3940b..4fce750435 100644
--- a/library/spdm_requester_lib/CMakeLists.txt
+++ b/library/spdm_requester_lib/CMakeLists.txt
@@ -32,3 +32,5 @@ SET(src_spdm_requester_lib
 )
 
 ADD_LIBRARY(spdm_requester_lib STATIC ${src_spdm_requester_lib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:spdm_requester_lib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/library/spdm_responder_lib/CMakeLists.txt b/library/spdm_responder_lib/CMakeLists.txt
index ffefcc1d0e..50fb5627df 100644
--- a/library/spdm_responder_lib/CMakeLists.txt
+++ b/library/spdm_responder_lib/CMakeLists.txt
@@ -36,4 +36,6 @@ SET(src_spdm_responder_lib
     libspdm_rsp_chunk_get.c
 )
 
-ADD_LIBRARY(spdm_responder_lib STATIC ${src_spdm_responder_lib})
\ No newline at end of file
+ADD_LIBRARY(spdm_responder_lib STATIC ${src_spdm_responder_lib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:spdm_responder_lib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
\ No newline at end of file
diff --git a/library/spdm_secured_message_lib/CMakeLists.txt b/library/spdm_secured_message_lib/CMakeLists.txt
index 0fa7cdd3a3..88b8b5ed61 100644
--- a/library/spdm_secured_message_lib/CMakeLists.txt
+++ b/library/spdm_secured_message_lib/CMakeLists.txt
@@ -10,3 +10,5 @@ SET(src_spdm_secured_message_lib
 )
 
 ADD_LIBRARY(spdm_secured_message_lib STATIC ${src_spdm_secured_message_lib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:spdm_secured_message_lib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/library/spdm_transport_mctp_lib/CMakeLists.txt b/library/spdm_transport_mctp_lib/CMakeLists.txt
index da6a29d313..e9e3ecaff4 100644
--- a/library/spdm_transport_mctp_lib/CMakeLists.txt
+++ b/library/spdm_transport_mctp_lib/CMakeLists.txt
@@ -8,3 +8,5 @@ SET(src_spdm_transport_mctp_lib
 )
 
 ADD_LIBRARY(spdm_transport_mctp_lib STATIC ${src_spdm_transport_mctp_lib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:spdm_transport_mctp_lib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/library/spdm_transport_pcidoe_lib/CMakeLists.txt b/library/spdm_transport_pcidoe_lib/CMakeLists.txt
index af7121f261..74690be198 100644
--- a/library/spdm_transport_pcidoe_lib/CMakeLists.txt
+++ b/library/spdm_transport_pcidoe_lib/CMakeLists.txt
@@ -8,3 +8,5 @@ SET(src_spdm_transport_pcidoe_lib
 )
 
 ADD_LIBRARY(spdm_transport_pcidoe_lib STATIC ${src_spdm_transport_pcidoe_lib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:spdm_transport_pcidoe_lib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/os_stub/cryptlib_openssl/CMakeLists.txt b/os_stub/cryptlib_openssl/CMakeLists.txt
old mode 100644
new mode 100755
index 1d246712f3..71856b788b
--- a/os_stub/cryptlib_openssl/CMakeLists.txt
+++ b/os_stub/cryptlib_openssl/CMakeLists.txt
@@ -21,6 +21,8 @@ elseif(ARCH STREQUAL "riscv32")
     ADD_COMPILE_OPTIONS(-DLIBSPDM_CPU_RISCV32)
 elseif(ARCH STREQUAL "riscv64")
     ADD_COMPILE_OPTIONS(-DLIBSPDM_CPU_RISCV64)
+elseif (ARCH STREQUAL "arm")
+    ADD_COMPILE_OPTIONS(-DLIBSPDM_CPU_ARM)
 else()
     MESSAGE(FATAL_ERROR "Unknown ARCH")
 endif()
@@ -57,3 +59,5 @@ endif()
 
 ADD_LIBRARY(cryptlib_openssl STATIC ${src_cryptlib_openssl})
 TARGET_COMPILE_OPTIONS(cryptlib_openssl PRIVATE ${OPENSSL_FLAGS})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:cryptlib_openssl>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/os_stub/debuglib/CMakeLists.txt b/os_stub/debuglib/CMakeLists.txt
index 80fbde1415..4f9d6ffd7a 100644
--- a/os_stub/debuglib/CMakeLists.txt
+++ b/os_stub/debuglib/CMakeLists.txt
@@ -8,3 +8,7 @@ SET(src_debuglib
 )
 
 ADD_LIBRARY(debuglib STATIC ${src_debuglib})
+if(NOT CMAKE_BUILD_TYPE STREQUAL "Release")
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:debuglib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
+endif()
diff --git a/os_stub/debuglib_null/CMakeLists.txt b/os_stub/debuglib_null/CMakeLists.txt
index cc23febf96..0ac6db8526 100644
--- a/os_stub/debuglib_null/CMakeLists.txt
+++ b/os_stub/debuglib_null/CMakeLists.txt
@@ -8,3 +8,7 @@ SET(src_debuglib_null
 )
 
 ADD_LIBRARY(debuglib_null STATIC ${src_debuglib_null})
+if(CMAKE_BUILD_TYPE STREQUAL "Release")
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:debuglib_null>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
+endif()
diff --git a/os_stub/malloclib/CMakeLists.txt b/os_stub/malloclib/CMakeLists.txt
index 35d8582908..d4226ea2eb 100644
--- a/os_stub/malloclib/CMakeLists.txt
+++ b/os_stub/malloclib/CMakeLists.txt
@@ -8,3 +8,5 @@ SET(src_malloclib
 )
 
 ADD_LIBRARY(malloclib STATIC ${src_malloclib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:malloclib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/os_stub/memlib/CMakeLists.txt b/os_stub/memlib/CMakeLists.txt
index 7632b2e726..1064916a35 100644
--- a/os_stub/memlib/CMakeLists.txt
+++ b/os_stub/memlib/CMakeLists.txt
@@ -11,3 +11,5 @@ SET(src_memlib
 )
 
 ADD_LIBRARY(memlib STATIC ${src_memlib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:memlib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/os_stub/openssllib/CMakeLists.txt b/os_stub/openssllib/CMakeLists.txt
index bd57bfb254..c0c76231cd 100644
--- a/os_stub/openssllib/CMakeLists.txt
+++ b/os_stub/openssllib/CMakeLists.txt
@@ -22,6 +22,8 @@ elseif(ARCH STREQUAL "riscv32")
     ADD_COMPILE_OPTIONS(-DLIBSPDM_CPU_RISCV32)
 elseif(ARCH STREQUAL "riscv64")
     ADD_COMPILE_OPTIONS(-DLIBSPDM_CPU_RISCV64)
+elseif (ARCH STREQUAL "arm")
+    ADD_COMPILE_OPTIONS(-DLIBSPDM_CPU_ARM)
 else()
     MESSAGE(FATAL_ERROR "Unknown ARCH")
 endif()
@@ -754,3 +756,5 @@ if ((ARCH STREQUAL "arm") OR (ARCH STREQUAL "aarch64"))
 endif()
 
 ADD_LIBRARY(openssllib STATIC ${src_openssllib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:openssllib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
\ No newline at end of file
diff --git a/os_stub/platform_lib/CMakeLists.txt b/os_stub/platform_lib/CMakeLists.txt
index 880e9396a6..ddf72c6489 100644
--- a/os_stub/platform_lib/CMakeLists.txt
+++ b/os_stub/platform_lib/CMakeLists.txt
@@ -21,3 +21,5 @@ SET(src_platform_lib
 endif()
 
 ADD_LIBRARY(platform_lib STATIC ${src_platform_lib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:platform_lib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/os_stub/rnglib/CMakeLists.txt b/os_stub/rnglib/CMakeLists.txt
old mode 100644
new mode 100755
index a053d71428..6a4b1bd576
--- a/os_stub/rnglib/CMakeLists.txt
+++ b/os_stub/rnglib/CMakeLists.txt
@@ -20,3 +20,5 @@ SET(src_rnglib
 endif()
 
 ADD_LIBRARY(rnglib STATIC ${src_rnglib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:rnglib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/os_stub/spdm_crypt_ext_lib/CMakeLists.txt b/os_stub/spdm_crypt_ext_lib/CMakeLists.txt
index 825ddd3646..30424b1be4 100644
--- a/os_stub/spdm_crypt_ext_lib/CMakeLists.txt
+++ b/os_stub/spdm_crypt_ext_lib/CMakeLists.txt
@@ -9,3 +9,5 @@ SET(src_spdm_crypt_ext_lib
 )
 
 ADD_LIBRARY(spdm_crypt_ext_lib STATIC ${src_spdm_crypt_ext_lib})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:spdm_crypt_ext_lib>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
diff --git a/os_stub/spdm_device_secret_lib_sample/CMakeLists.txt b/os_stub/spdm_device_secret_lib_sample/CMakeLists.txt
index 0900a58a7e..5ac61ab188 100644
--- a/os_stub/spdm_device_secret_lib_sample/CMakeLists.txt
+++ b/os_stub/spdm_device_secret_lib_sample/CMakeLists.txt
@@ -16,3 +16,6 @@ if ((ARCH STREQUAL "arm") OR (ARCH STREQUAL "aarch64"))
 endif()
 
 ADD_LIBRARY(spdm_device_secret_lib_sample STATIC ${src_spdm_device_secret_lib_sample})
+list(APPEND SPDM_OBJECTS $<TARGET_OBJECTS:spdm_device_secret_lib_sample>)
+set(SPDM_OBJECTS ${SPDM_OBJECTS} PARENT_SCOPE)
+
-- 
2.34.1

