From 72c66fc8c380a62e6b80d515ef3e0a3281132bca Mon Sep 17 00:00:00 2001
From: Eddyhan <eddyhan@ami.com>
Date: Mon, 27 Jun 2022 12:10:54 +0800
Subject: [PATCH] Add libspdm.pc.in


diff --git a/libspdm.pc.in b/libspdm.pc.in
new file mode 100755
index 00000000..3c909a79
--- /dev/null
+++ b/libspdm.pc.in
@@ -0,0 +1,11 @@
+prefix="@CMAKE_INSTALL_PREFIX@"
+exec_prefix="${prefix}"
+libdir="${prefix}/@CMAKE_INSTALL_LIBDIR@"
+includedir="${prefix}/include/"
+
+Name: @PROJECT_NAME@
+Description: @PROJECT_DESCRIPTION@
+Version: @PROJECT_VERSION@
+Requires: @PKGCONF_REQ_PUB@
+Cflags: -I"${includedir}" -I"${includedir}/libspdm/include" -I"${includedir}/libspdm/include/hal" -I"${includedir}/libspdm/os_stub" -I"${includedir}libspdm/os_stub/spdm_crypt_ext_lib" -I"${includedir}/libspdm/os_stub/include"
+Libs: -L"${libdir}" -l@PROJECT_NAME@
-- 
2.34.1

