From 06f44f813e364a56db79418ad2a266100ec5cb46 Mon Sep 17 00:00:00 2001
From: sivanagalakshmi <sivanagalakshmia@ami.com>
Date: Thu, 12 Dec 2024 03:47:12 -0500
Subject: [PATCH] Added_peci_wakeup_support_for_lib_peci

---
 peci.c | 24 ++++++++++++++++++++++++
 peci.h |  2 +-
 2 files changed, 25 insertions(+), 1 deletion(-)

diff --git a/peci.c b/peci.c
index 9c984a8..27aa5f1 100644
--- a/peci.c
+++ b/peci.c
@@ -2445,3 +2445,27 @@ EPECIStatus peci_GetCPUID(const uint8_t clientAddr, CPUModel* cpuModel,
     *stepping = (uint8_t)(cpuid & 0x0000000F);
     return ret;
 }
+
+/*-------------------------------------------------------------------------
+ * This function allows to wake up the peci
+ *------------------------------------------------------------------------*/
+EPECIStatus peci_WakePECI(const uint8_t clientAddr, uint8_t domainId, uint8_t* cc)
+{
+        EPECIStatus ret     = PECI_CC_SUCCESS;
+
+        uint8_t  pkgIndex   = PECI_MBX_INDEX_WAKE_MODE_BIT;
+        uint16_t param      = 2;
+        uint8_t  wrLen      = sizeof(uint32_t);
+        uint32_t value      = 0;
+
+        // The client address must be in the valid range
+        if (clientAddr < MIN_CLIENT_ADDR || clientAddr > MAX_CLIENT_ADDR)
+        {
+                return PECI_CC_INVALID_REQ;
+        }
+
+        ret = peci_WrPkgConfig_dom(clientAddr,domainId, pkgIndex, param, &value, wrLen, cc);
+
+        return ret;
+}
+
diff --git a/peci.h b/peci.h
index a012771..869d42e 100644
--- a/peci.h
+++ b/peci.h
@@ -501,7 +501,7 @@ EPECIStatus peci_GetCPUID(const uint8_t clientAddr, CPUModel* cpuModel,
                           uint8_t* stepping, uint8_t* cc);
 void peci_SetDevName(char* peci_dev);
 int peci_i3c_chardev_to_cpu(const char* name);
-
+EPECIStatus peci_WakePECI(const uint8_t clientAddr, uint8_t domainId, uint8_t* cc);
 #ifdef __cplusplus
 }
 #endif
-- 
2.34.1

