From cd12fcc963bdb0e6909989665deb1960878d4d3b Mon Sep 17 00:00:00 2001
From: sujithm <sujithm@ami.com>
Date: Thu, 30 Jan 2025 20:10:49 +0530
Subject: [PATCH 1/1] Added-coverity-changes-for-peci

---
 peci.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/peci.c b/peci.c
index 9c984a8..4acc595 100644
--- a/peci.c
+++ b/peci.c
@@ -173,6 +173,11 @@ int peci_i3c_chardev_to_cpu(const char* name)
     {
         goto err_free;
     }
+    
+    if (strstr(lpath,"peci-i3c") == NULL )
+    { 
+	return -1
+    }
 
     subpath = strstr(lpath, "peci-i3c");
     if (!subpath)
-- 
2.43.0

