From 3ef2f162992a26c1e1c311cace62dc69a5c5d5ba Mon Sep 17 00:00:00 2001
From: arjuncr <arjuncr@ami.com>
Date: Wed, 23 Jul 2025 12:39:37 -0400
Subject: [PATCH] fix for cups context switch

---
 peci.c | 14 +++++++-------
 peci.h |  4 +++-
 2 files changed, 10 insertions(+), 8 deletions(-)

diff --git a/peci.c b/peci.c
index 4b56512..2e74e41 100644
--- a/peci.c
+++ b/peci.c
@@ -63,14 +63,14 @@ void peci_SetDevName(char* peci_dev)
         peci_name_new[DEV_NAME_SIZE - 1] = '\0';
         peci_device_list[0] = peci_name_new;
         peci_device_list[1] = NULL;
-        syslog(LOG_INFO, "PECI set dev name to %s\n", peci_device_list[0]);
+        //syslog(LOG_INFO, "PECI set dev name to %s\n", peci_device_list[0]);
     }
     else
     {
         peci_device_list[0] = "/dev/peci-default";
         peci_device_list[1] = "/dev/peci-0";
-        syslog(LOG_INFO, "PECI set dev names to %s, %s\n", peci_device_list[0],
-               peci_device_list[1]);
+        //syslog(LOG_INFO, "PECI set dev names to %s, %s\n", peci_device_list[0],
+        //       peci_device_list[1]);
     }
 }
 
@@ -368,7 +368,7 @@ EPECIStatus peci_Lock(int* peci_fd, int timeout_ms)
 /*-------------------------------------------------------------------------
  * This function closes the peci interface
  *------------------------------------------------------------------------*/
-static void peci_Close(int peci_fd)
+void peci_Close(int peci_fd)
 {
     peci_Unlock(peci_fd);
 }
@@ -376,7 +376,7 @@ static void peci_Close(int peci_fd)
 /*-------------------------------------------------------------------------
  * This function opens the peci interface and returns a file descriptor
  *------------------------------------------------------------------------*/
-static EPECIStatus peci_Open(int* peci_fd)
+EPECIStatus peci_Open(int* peci_fd)
 {
     if (NULL == peci_fd)
     {
@@ -2449,7 +2449,7 @@ EPECIStatus peci_GetCPUID(const uint8_t clientAddr, CPUModel* cpuModel,
 /*-------------------------------------------------------------------------
  * This function allows to wake up the peci
  *------------------------------------------------------------------------*/
-EPECIStatus peci_WakePECI(const uint8_t clientAddr, uint8_t domainId, uint8_t* cc)
+EPECIStatus peci_WakePECI(const uint8_t clientAddr, uint8_t domainId, int peci_fd, uint8_t* cc)
 {
         EPECIStatus ret     = PECI_CC_SUCCESS;
 
@@ -2464,7 +2464,7 @@ EPECIStatus peci_WakePECI(const uint8_t clientAddr, uint8_t domainId, uint8_t* c
                 return PECI_CC_INVALID_REQ;
         }
 
-        ret = peci_WrPkgConfig_dom(clientAddr,domainId, pkgIndex, param, &value, wrLen, cc);
+        ret = peci_WrPkgConfig_seq_dom(clientAddr,domainId, pkgIndex, param, &value, wrLen, peci_fd, cc);
 
         return ret;
 }
diff --git a/peci.h b/peci.h
index 869d42e..d8d447d 100644
--- a/peci.h
+++ b/peci.h
@@ -501,7 +501,9 @@ EPECIStatus peci_GetCPUID(const uint8_t clientAddr, CPUModel* cpuModel,
                           uint8_t* stepping, uint8_t* cc);
 void peci_SetDevName(char* peci_dev);
 int peci_i3c_chardev_to_cpu(const char* name);
-EPECIStatus peci_WakePECI(const uint8_t clientAddr, uint8_t domainId, uint8_t* cc);
+void peci_Close(int peci_fd);
+EPECIStatus peci_Open(int* peci_fd);
+EPECIStatus peci_WakePECI(const uint8_t clientAddr, uint8_t domainId, int peci_fd, uint8_t* cc);
 #ifdef __cplusplus
 }
 #endif
-- 
2.34.1

