From ea08602291fe824d89dd8a5885cb08d80b23ae76 Mon Sep 17 00:00:00 2001
From: gayathrid <gayathrid@ami.com>
Date: Tue, 16 Sep 2025 02:12:54 -0400
Subject: [PATCH] makefile and kconfig changes for peci client

---
 drivers/mfd/Kconfig  | 18 ++++++++++++++++++
 drivers/mfd/Makefile |  1 +
 2 files changed, 19 insertions(+)

diff --git a/drivers/mfd/Kconfig b/drivers/mfd/Kconfig
index 68d71b4b55bd..d04514ba3594 100644
--- a/drivers/mfd/Kconfig
+++ b/drivers/mfd/Kconfig
@@ -700,6 +700,24 @@ config MFD_INTEL_LPSS_PCI
 	  I2C, SPI and HS-UART starting from Intel Sunrisepoint (Intel Skylake
 	  PCH) in PCI mode.
 
+config MFD_INTEL_PECI_CLIENT
+        tristate "Intel PECI client"
+        depends on (PECI_LEGACY || COMPILE_TEST)
+        select MFD_CORE
+        help
+          If you say yes to this option, support will be included for the
+          Intel PECI (Platform Environment Control Interface) client. PECI is a
+          one-wire bus interface that provides a communication channel from PECI
+          clients in Intel processors and chipset components to external
+          monitoring or control devices.
+
+          Additional drivers must be enabled in order to use the functionality
+          of the device.
+
+          This driver can also be built as a module. If so, the module
+          will be called intel-peci-client.
+
+
 config MFD_INTEL_PMC_BXT
 	tristate "Intel PMC Driver for Broxton"
 	depends on X86
diff --git a/drivers/mfd/Makefile b/drivers/mfd/Makefile
index db1ba39de3b5..c603657a81c2 100644
--- a/drivers/mfd/Makefile
+++ b/drivers/mfd/Makefile
@@ -282,3 +282,4 @@ obj-$(CONFIG_MFD_ATC260X_I2C)	+= atc260x-i2c.o
 
 obj-$(CONFIG_MFD_RSMU_I2C)	+= rsmu_i2c.o rsmu_core.o
 obj-$(CONFIG_MFD_RSMU_SPI)	+= rsmu_spi.o rsmu_core.o
+obj-$(CONFIG_MFD_INTEL_PECI_CLIENT)     += intel-peci-client.o
-- 
2.34.1

