From 3e52e47fb30fa95dbd7b7636ccd55d479ab726f2 Mon Sep 17 00:00:00 2001
From: arjuncr <arjuncr@ami.com>
Date: Fri, 19 Sep 2025 03:59:49 -0400
Subject: [PATCH] fix for lpc sio driver not loading

---
 drivers/soc/aspeed/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/aspeed/Kconfig b/drivers/soc/aspeed/Kconfig
index 6460695108f4..ba48a25aa741 100644
--- a/drivers/soc/aspeed/Kconfig
+++ b/drivers/soc/aspeed/Kconfig
@@ -21,7 +21,7 @@ config ASPEED_LPC_CTRL
 	  host LPC read/write region can be buffered.
 
 config ASPEED_LPC_SIO
-	tristate "Aspeed ast2400/2500 HOST LPC SIO support"
+	tristate "Aspeed ast2600 HOST LPC SIO support"
 	depends on REGMAP && MFD_SYSCON
 	help
 	  Provides a driver to control the LPC SIO interface on ASPEED platform
-- 
2.34.1

