From 9ac1e17dcbde7b6f05dee24dc8e63837c9361db5 Mon Sep 17 00:00:00 2001
From: sujithm <sujithm@ami.com>
Date: Wed, 17 Dec 2025 20:56:49 +0530
Subject: [PATCH] kernel coverity fix

---
 drivers/soc/aspeed/aspeed-espi-oob.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/soc/aspeed/aspeed-espi-oob.c b/drivers/soc/aspeed/aspeed-espi-oob.c
index ca9c362c897d..7c3ca5afafd8 100644
--- a/drivers/soc/aspeed/aspeed-espi-oob.c
+++ b/drivers/soc/aspeed/aspeed-espi-oob.c
@@ -418,6 +418,7 @@ static int aspeed_espi_oob_dma_alloc(struct device *dev, struct aspeed_espi_oob
 			return -ENOMEM;
 		}
 
+		/* coverity[double_unlock : FALSE] */
 		dma->rx_desc = dmam_alloc_coherent(dev, sizeof(*dma->rx_desc) * dma->rx_desc_num,
 						   &dma->rx_desc_addr, GFP_KERNEL);
 		if (!dma->rx_desc) {
-- 
2.34.1

