diff --git a/drivers/jtag/jtag.c b/drivers/jtag/jtag.c
--- a/drivers/jtag/jtag.c
+++ b/drivers/jtag/jtag.c
@@ -289,8 +289,10 @@ struct jtag *jtag_alloc(struct device *host, size_t priv_size,
 	if (!jtag)
 		return NULL;
 	jtag->priv = kzalloc(priv_size, GFP_KERNEL);
-	if (!jtag->priv)
+	if (!jtag->priv) {
+        jtag->priv = NULL;
 		return NULL;
+    }
 
 	jtag->ops = ops;
 	jtag->miscdev.parent = host;
